### PR TITLE
Added Sqrt monotonic binary search in cpp

### PR DESCRIPTION
**Pull request by satcasm**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have added  Sqrt monotonic binary search in cpp

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#5418

### CODE REVIEW

**Comment by satcasm** [comment]
@ankitaggarwal23 @HarshCasper please review my PR.

File: C-Plus-Plus/search/Sqrt_Monotonic_Binary_Search.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// In this algorithm we will find the square root of a number `n` upto `p` digits after decimal using the binary search algorithm.
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/search/Sqrt_Monotonic_Binary_Search.cpp]: Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top

File: C-Plus-Plus/search/Sqrt_Monotonic_Binary_Search.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+#include <iostream>
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/search/Sqrt_Monotonic_Binary_Search.cpp]: replace header with `#include <bits/stdc++.h>`

File: C-Plus-Plus/search/Sqrt_Monotonic_Binary_Search.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+using namespace std;
+
+// Function to find the square root of a number `n`
+float sqrt(int n, int p)
+{
+	int s = 0;		// s denotes the starting point
+	int e = n;		// e denotes the end point
+	float ans = -1; // ans will store the final answer
+	int mid;		// mid will store the middle element
+	while (s <= e)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/search/Sqrt_Monotonic_Binary_Search.cpp]: remove redundant comments and use proper naming conventions for variables

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/search/Sqrt_Monotonic_Binary_Search.cpp (hunk 2 above)
Comment (satcasm) [reply C-Plus-Plus/search/Sqrt_Monotonic_Binary_Search.cpp]: Done.

**Comment by satcasm** [comment]
Hey @ankitaggarwal23  made the asked changes, kindly have a review.
Thanks!

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**